airflow/models/dag.py

```diff
@@ -2347,6 +2352,19 @@ def create_dagrun(
             creating_job_id=creating_job_id,
             data_interval=data_interval,
         )
+
+        # prevent triggering manually for scheduled DAGs, while 'allow_manual' = False
+        if (
+            run.run_type == DagRunType.MANUAL
+            and self.schedule_interval is not None
+            and self.schedule_interval != "@once"
+            and not self.allow_manual
+        ):
+            raise AirflowException(
+                f"Cannot manually trigger DAG {self.dag_id}, because it is a scheduled DAG, "
+                f"and 'allow_manual' is set to {self.allow_manual}"
```

Comment:
Can we add a unittest so that this is not accidentally removed in the future

Reply:
I have added a test case 👍